From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 02 Feb 2022 00:55:24 +0100
Subject: [PATCH] Change runtime directory location to follow FHS

Helix allows to specify the location of the runtime directory (which
contains arch-independent modules) via environment variable
`HELIX_RUNTIME`. Using this would require installing a wrapper script
to set this variable prior executing `hx`.

If this environment variable is not set, Helix fallbacks to directory
`./runtime` relative to the directory of the `hx` executable. This is not
suitable for Alpine (FHS).

This patch changes the fallback to `/usr/share/helix/runtime`.

--- a/helix-core/src/lib.rs
+++ b/helix-core/src/lib.rs
@@ -100,11 +100,13 @@
         return std::path::PathBuf::from(dir).parent().unwrap().join(RT_DIR);
     }
 
+    // XXX-Patched
+    return std::path::PathBuf::from("/usr/share/helix").join(RT_DIR);
     // fallback to location of the executable being run
-    std::env::current_exe()
-        .ok()
-        .and_then(|path| path.parent().map(|path| path.to_path_buf().join(RT_DIR)))
-        .unwrap()
+    //std::env::current_exe()
+    //    .ok()
+    //    .and_then(|path| path.parent().map(|path| path.to_path_buf().join(RT_DIR)))
+    //    .unwrap()
 }
 
 pub fn config_dir() -> std::path::PathBuf {
